src/split-props.js

```diff
@@ -1,4 +1,35 @@
+// @flow
+import type {Component} from 'react'
 import shouldForwardProperty from './should-forward-property'
+import type {
+  FunctionalComponent,
+  ComponentClass,
+  RefCallback,
+} from './types/React'
+import type {CSSProperties} from './types/CSSProperties'
+import type {Theme} from './theme-provider'
+
+export type SplittableProps = {
+  css: CSSProperties,
+  theme: Theme,
+  className: string,
+  // TODO: Use rootEl to give a more specific typing for this innerRef's
```

Comment:
Ah, this isn't necessarily important, but I was saying that instead of typing innerRef as `RefCallback<Component<*, *, *>>` (aka ref callback of any component instance), if rootEl is set to eg. "div", it could be `RefCallback<HTMLDivElement>`.

Looking at it again, unless I add in some knowledge of when it'll be a component instance and when it'll be an html node, this should probably use `RefCallback<any>` instead, because otherwise the type will be incorrect when the ref is for an html element.